### PR TITLE
Corrects `logs` command to validate `--limit` option's value if it is <=0

### PR DESCRIPTION
**Pull request by anshulvermapatel**
Resolves [#909](https://github.com/tektoncd/cli/issues/909)

# Changes

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [ ] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality changed/added)
- [ ] Run the code checkers with `make check`
- [ ] Regenerate the manpages, docs and go formatting with `make generated`
- [ ] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/cli/blob/master/CONTRIBUTING.md)
for more details._

# Release Notes

```
release-note
```

### CODE REVIEW

File: pkg/cmd/pipelinerun/logs.go [hunk 1]
```diff
@@ -91,6 +91,10 @@ Show the logs of PipelineRun named 'microservice-1' for all tasks and steps (inc
 }
 
 func Run(opts *options.LogOptions) error {
+	if err := opts.ValidateOpts(); err != nil {
```
Comment (danielhelfand) [inline pkg/cmd/pipelinerun/logs.go:94]: Please add a unit test for this case.

File: pkg/cmd/taskrun/logs.go [hunk 2]
```diff
@@ -94,6 +94,10 @@ Show the logs of TaskRun named 'microservice-1' for step 'build' only from names
 }
 
 func Run(opts *options.LogOptions) error {
+	if err := opts.ValidateOpts(); err != nil {
```
Comment (danielhelfand) [inline pkg/cmd/taskrun/logs.go:97]: Please add a unit test for this case.

**Comment by danielhelfand** [comment]
The unit test failures are related to not updating existing unit tests with a defaultLimit value for logOptions:

https://github.com/tektoncd/cli/blob/2d9aad837e7d1380da8a6e5a267bdc1ff57d0185/pkg/cmd/pipelinerun/logs_test.go#L1057

File: pkg/cmd/pipelinerun/logs_test.go [hunk 3]
```diff
@@ -2095,6 +2120,7 @@ func logOpts(name string, ns string, cs pipelinetest.Clients, dc dynamic.Interfa
 }
 
 func fetchLogs(lo *options.LogOptions) (string, error) {
+	fmt.Println("Limit is -", lo.Limit)
```
Comment (danielhelfand) [inline pkg/cmd/pipelinerun/logs_test.go:2123]: This looks like it was from debugging. Please remove it.

File: pkg/cmd/pipelinerun/logs_test.go [hunk 4]
```diff
@@ -277,7 +302,7 @@ func TestPipelinerunLogs(t *testing.T) {
 			),
 		),
 	}
-
+	fmt.Println("PipelineRuns are -", prs) // Mine
```
Comment (danielhelfand) [inline pkg/cmd/pipelinerun/logs_test.go:305]: This looks like it was from debugging. Please remove it.

File: pkg/cmd/pipelinerun/logs_test.go [hunk 5]
```diff
@@ -16,6 +16,7 @@ package pipelinerun
 
 import (
 	"bytes"
+	"fmt"
```
Comment (danielhelfand) [inline pkg/cmd/pipelinerun/logs_test.go:19]: Remove this import

File: pkg/cmd/pipelinerun/logs_test.go (hunk 4 above)
Comment (anshulvermapatel) [reply pkg/cmd/pipelinerun/logs_test.go]: I am sorry about that. I have made the correction.

File: pkg/cmd/pipelinerun/logs_test.go (hunk 5 above)
Comment (anshulvermapatel) [reply pkg/cmd/pipelinerun/logs_test.go]: Done

File: pkg/cmd/pipelinerun/logs_test.go (hunk 3 above)
Comment (anshulvermapatel) [reply pkg/cmd/pipelinerun/logs_test.go]: Removed it, Sorry about that.

**Pull request merged**